addons/pos_hr/static/src/js/CashierName.js

```diff
@@ -16,7 +16,7 @@ odoo.define('pos_hr.CashierName', function (require) {
             get avatar() {
                 if (this.env.pos.config.module_pos_hr) {
                     const cashier = this.env.pos.get_cashier();
-                    return `/web/image/hr.employee/${cashier.id}/avatar_128`;
+                    return `/web/image/hr.employee.public/${cashier.id}/avatar_128`;
```

Comment:
I'm not so sure if hr.employee.public records points to the same hr.employee records. From a quick check, I think the id needs to be extracted from "hr.employee.public".employee_id. But I'm not so sure.

Reply:
Thank you for reviewing the PR

I think not because by using hr.employee.public model we can access the public data of the employee (hr.employee) but we can't modify it.
https://github.com/odoo/odoo/commit/c9ca3761464413327d2beb697553a3ccd7eef4d1

Reply:
I check and I found that our use of "hr.employee" is not the issue. The problem is because we try to fetch image of an hr.employee with "id = null". It's the same problem even if we use the public version of the model. Check the following image, even with your fix, we still get the 404 error.

<img width="763" alt="Screenshot 2023-04-06 at 19 12 30" src="https://user-images.githubusercontent.com/3245568/230448976-73392f25-ca83-4a2b-b5fa-15a46ca9e109.png">

In my opinion, the proper fix is to load a fallback image when the id is null.

Reply:
Yes, Indeed, I also got the same error locally now.
But now, It is fixed